conda_smithy/cli.py

```diff
@@ -192,16 +192,52 @@ def __init__(self, parser):
         scp = self.subcommand_parser
         scp.add_argument("--feedstock_directory", default=os.getcwd(),
                          help="The directory of the feedstock git repository.")
+        scp.add_argument("-c", "--commit", nargs='?', choices=["edit", "auto"], const="edit",
+                         help="Whether to setup a commit or not.")
 
     def __call__(self, args):
         try:
             configure_feedstock.main(args.feedstock_directory)
-            print("\nCI support files regenerated. These need to be pushed to github!")
-            print("\nYou can commit the changes with:\n\n"
-                  "    git add -A\n"
-                  "    git commit -m 'MNT: Re-rendered with conda-smithy %s'" % __version__)
+            print("\nRe-rendered with conda-smithy %s'." % __version__)
+
+            is_git_repo = os.path.exists(os.path.join(args.feedstock_directory, ".git"))
+            if is_git_repo:
+                has_staged_changes = subprocess.call(
+                    [
+                        "git", "diff", "--cached", "--quiet", "--exit-code"
+                    ],
+                    cwd=args.feedstock_directory
+                )
+                if has_staged_changes:
+                    if args.commit:
+                        git_args = [
+                            'git',
+                            'commit',
+                            '-m',
+                            'MNT: Re-rendered with conda-smithy %s' % __version__
+                        ]
+                        if args.commit == "edit":
+                            git_args += [
+                                '--edit',
+                                '--status',
+                                '--verbose'
+                            ]
+                        subprocess.check_call(
+                            git_args,
+                            cwd=args.feedstock_directory
+                        )
+                    else:
+                        print(
+                            "\nYou can commit the changes with:\n\n"
+                            "    git commit -m 'MNT: Re-rendered with conda-smithy %s'\n" % __version__
+                        )
+                    print("\nThese changes need to be pushed to github!\n")
+                else:
+                    print("\nNo changes made. This feedstock is up-to-date.\n")
```

Comment:
Made some spacing adjustments with these `print` statements. Seems there were a couple cases that got a bit clumped together. PR ( https://github.com/conda-forge/conda-smithy/pull/343 ) seems to fix this issue.